sklearn_numba_dpex/kmeans/drivers.py

```diff
@@ -740,8 +779,10 @@ def kmeans_plusplus(
     candidate_ids = dpt.empty(sh=(n_local_trials,), dtype=np.int32, device=device)
 
     # Pick first center randomly
-    starting_center_id = np.int32(get_random_raw(random_state) % n_samples)
-    center_indices[0] = starting_center_id[0]
+    starting_center_id = dpt.asnumpy(get_random_raw(random_state))[0] % np.uint64(
+        n_samples
+    )
+    center_indices[0] = np.int32(starting_center_id)
```

Comment:
Shouldn't it be something like this, here?
```suggestion
    center_indices[0] = np.int32(starting_center_id[0])
```

Reply:
OK, let's break it in two pieces if this LGTY. :)